File: public/dsa/searching/contains_element/naive/contains_element.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    bool containsElement(vector<int> &arr, int key) {
        int n = arr.size();
        for (int i = 0; i < n; i++) {
            if (arr[i] == key) {
                return true;
            }
        }
        return false;
    }
};

// tests-start
int main() {
    vector<int> arr = {1, 2, 3, 3, 3, 4, 4, 5};
    int key = 2;

    Solution sol;
    cout << (sol.containsElement(arr, key) ? "true" : "false") << endl;

    return 0;
}
// tests-end

/*output
true
*/

File: public/dsa/searching/contains_element/naive/contains_element.py
from typing import List


class Solution:
    def containsElement(self, arr: List[int], key: int) -> bool:
        n = len(arr)
        for i in range(n):
            if arr[i] == key:
                return True
        return False


# tests-start
arr = [1, 2, 3, 3, 3, 4, 4, 5]
key = 2
sol = Solution()
print(sol.containsElement(arr, key))
# tests-end

'''output
True
'''
